packages/jest-cli/src/run_jest.js

```diff
@@ -75,6 +75,12 @@ const processResults = (runResults, options) => {
   return options.onComplete && options.onComplete(runResults);
 };
 
+// eslint-disable-next-line prefer-const
+let testSchedulerContext = {
+  firstRun: true,
+  previousSuccess: true,
+};
+
```

Comment:
Let me know what you think of this. I wasn't sure what the best practice for updating the state from my previous run in NotifyReporter would be.

Reply:
The `NotifyReporter` can have local state, can't it? Why not just have `this._firstRun = false` etc in there?

Reply:
run_jest seems to create a new instance of TestScheduler every time a file changes with `--watch` and TestScheduler creates a new instance of NotifyReporter.